substrate/frame/fast-unstake/src/lib.rs

```diff
@@ -558,26 +576,37 @@ pub mod pallet {
 				unchecked_eras_to_check
 			);
 
-			let unstake_stash = |stash: T::AccountId, deposit| {
+			let unstake_stash = |stash: T::AccountId| {
 				let result = T::Staking::force_unstake(stash.clone());
-				let remaining = T::Currency::unreserve(&stash, deposit);
-				if !remaining.is_zero() {
+				let release_outcome = T::Currency::release_all(
+					&HoldReason::FastUnstake.into(),
+					&stash,
+					Precision::BestEffort,
+				);
+				if release_outcome.is_err() {
 					Self::halt("not enough balance to unreserve");
 				} else {
 					log!(info, "unstaked {:?}, outcome: {:?}", stash, result);
 					Self::deposit_event(Event::<T>::Unstaked { stash, result });
 				}
 			};
 
-			let check_stash = |stash, deposit| {
+			let check_stash = |stash| {
 				let is_exposed = unchecked_eras_to_check
 					.iter()
 					.any(|e| T::Staking::is_exposed_in_era(&stash, e));
 
 				if is_exposed {
-					T::Currency::slash_reserved(&stash, deposit);
-					log!(info, "slashed {:?} by {:?}", stash, deposit);
-					Self::deposit_event(Event::<T>::Slashed { stash, amount: deposit });
+					let _ = T::Currency::burn_all_held(
+						&HoldReason::FastUnstake.into(),
+						&stash,
+						Precision::BestEffort,
+						// This is not a system-level slash, so we politely do our best to slash.
+						Fortitude::Polite,
+					)
```

Comment:
`slash_reserved` created a negative imbalance, whereas `burn_all_held` just burns it. Don't we still need that negative imbalance?

Reply:
If we need to slash instead of burn, we might need a `slash_all_held`